### ISSUE TITLE
R-3.4.4: Checksum verification for extension source mgcv_1.8-23.tar.gz failed

### ISSUE THREAD

**Issue by tardigradus** [easybuilders/easybuild-easyconfigs#6501]
Building R from [here](https://github.com/easybuilders/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/r/R/R-3.4.4-foss-2018a-X11-20180131.eb) fails with the following error:

[xxx@centos7box EasyBuild]$ eb recipes/R-3.4.4-foss-2018a-X11-20180131.eb
== temporary log file in case of crash /tmp/eb-2JgDPj/easybuild-Aha9jU.log
== processing EasyBuild easyconfig /home/xxx/EasyBuild/recipes/R-3.4.4-foss-2018a-X11-20180131.eb
== building and installing R/3.4.4-foss-2018a-X11-20180131...
== fetching files...
== FAILED: Installation ended unsuccessfully (build directory: /home/xxx/.local/easybuild/build/R/3.4.4/foss-2018a-X11-20180131): build failed (first 300 chars): Checksum verification for extension source mgcv_1.8-23.tar.gz failed
== Results of the build can be found in the log file(s) /tmp/eb-2JgDPj/easybuild-R-3.4.4-20180627.142018.xpLnL.log
ERROR: Build of /home/xxx/EasyBuild/recipes/R-3.4.4-foss-2018a-X11-20180131.eb failed (err: 'build failed (first 300 chars): Checksum verification for extension source mgcv_1.8-23.tar.gz failed')

The solution given for #4208 is to dowload the package manually.  Is that the solution here, or can the easyconfig file be fixed?

**Comment by mstud** [comment]
I had the same problem, and I don't think downloading them manually is going to help. Problem is that the packages can change in CRAN while keeping the same version/filename, I believe. I resorted to using --ignore-checksums ...

**Comment by boegel** [comment]
Ugh, this seems to be occurring more & more... (see also #6446)

@tardigradus Which checksum are you getting now? Here's what I'm getting for a fresh download from CRAN via https://cran.r-project.org/src/contrib/Archive/mgcv/ (which means the original `mgcv_1.8-23.tar.gz` was replaced at some point):

```
$ sha256sum mgcv_1.8-23.tar.gz
a4d644943a2c91509f21f112fc5aca52f11af38f7bf9c987de4ce820918b2e6e  mgcv_1.8-23.tar.gz
```

The easy way out is indeed to use `--ignore-checksums`, but that defeats the purpose of having checksums in the first place.
You can also edit the easyconfig file to replace the faulty checksum...

@JoFAM Is there any way we can formally report this problem to the CRAN maintainers? This is getting really annoying...

**Comment by tardigradus** [comment]
In the log I have 

`== 2018-06-27 14:20:58,764 easyblock.py:510 INFO sha256 checksum for /home/loris/.local/easybuild/sources/r/R/extensions/mgcv_1.8-23.tar.gz: a4d644943a2c91509f21f112fc5aca52f11af38f7bf9c987de4ce820918b2e6e`

but the checksum in the easyconfig is

`7771ac5581542904d569388e2e498d3e3b8b65f7a81737acbb45d8ef838359d3`

**Comment by boegel** [comment]
@tardigradus Indeed, that's because the upstream checksum has changed for some reason...

Here's the diff between the old and the new source tarball:

```
[15:28:47] vsc40023@gligar02:~/data/EasyBuild/sources/r/R/extensions $ diff -ru mgcv.old mgcv.new
diff -ru mgcv.old/DESCRIPTION mgcv.new/DESCRIPTION
--- mgcv.old/DESCRIPTION	2018-01-15 01:23:36.000000000 +0100
+++ mgcv.new/DESCRIPTION	2018-01-21 19:59:15.000000000 +0100
@@ -20,4 +20,4 @@
 NeedsCompilation: yes
 Packaged: 2018-01-08 15:15:43 UTC; sw283
 Repository: CRAN
-Date/Publication: 2018-01-15 00:23:36 UTC
+Date/Publication: 2018-01-21 18:59:15 UTC
diff -ru mgcv.old/MD5 mgcv.new/MD5
--- mgcv.old/MD5	2018-01-15 01:23:36.000000000 +0100
+++ mgcv.new/MD5	2018-01-21 19:59:15.000000000 +0100
@@ -1,5 +1,5 @@
 6e3790e8df181acf2ecf469373521368 *ChangeLog
-a01208f1c73866153328f2ce262e28d9 *DESCRIPTION
+1a5c2e825ac38aca7a306bee8ce7b4c5 *DESCRIPTION
 eb723b61539feef013de476e68b5c50a *GPL-2
 20feb98661e2c2d10bf2d3337af57cb3 *NAMESPACE
 256ca3ddb2f799583b97d24ac136e579 *R/bam.r
```

It's unclear to me why the checksum only changed now (and now in Jan'18 as the diff suggests)...

**Comment by JoFAM** [comment]
@boegel I wish there was... R-devel and R-package-devel mailinglists are about the most official way I can contact CRAN. I've reported them the previous issues and one of the maintainers was going to look into it, but I haven't heard back yet. To their defense, they're seriously understaffed and swamped with work. 

The problem of changing publication date won't be solved anytime soon. I don't understand why they do that, but that happens often. The problem of files without any visible diff still having a different checksum, boggles me.

I asked again, will keep you updated. I'll also reach out to the community.

**Comment by boegel** [comment]
I re-downloaded all sources for the R 3.4.4 easyconfigs, seems like only `mgcv` is affected, fixed in #6502...

**Comment by migueldiascosta** [comment]
fixed in #6502
